internal/observability/event/event_type.go

```diff
@@ -24,3 +27,11 @@ func (et EventType) Validate() error {
 		return fmt.Errorf("%s: '%s' is not a valid event type: %w", op, et, ErrInvalidParameter)
 	}
 }
+
+// GenerateNodeID generates a new UUID.
+// It currently ignores any errors that may occur.
+func GenerateNodeID() eventlogger.NodeID {
+	id, _ := uuid.GenerateUUID()
```

Comment:
Are we positive this can never return an error?  Maybe a panic is in order?  Or just, you know, handle the error?

Reply:
This was an oversight. There is a very real possibility that an error could be returned here since the **uuid.GenerateUUID** function relies on a **rand.Reader**. I will fix this.